libraries/rush-lib/src/logic/operations/IPCOperationRunner.ts

```diff
@@ -143,11 +143,14 @@ export class IPCOperationRunner implements IOperationRunner {
             }
           }
 
-          function onExit(code: number): void {
+          function onExit(exitCode: number | null, signal: NodeJS.Signals | null): void {
             try {
-              if (code !== 0) {
+              if (exitCode !== 0) {
                 // Do NOT reject here immediately, give a chance for other logic to suppress the error
-                context.error = new OperationError('error', `Returned error code: ${code}`);
+                context.error = new OperationError('error', `Returned error code: ${exitCode}`);
+                resolve(OperationStatus.Failure);
+              } else if (signal) {
+                context.error = new OperationError('error', `Terminated by signal: ${signal}`);
                 resolve(OperationStatus.Failure);
```

Comment:
Flip the order of checks. `signal` being non-empty automatically means `exitCode === null` (match ShellOperaitonRunner's ordering).

Reply:
fixed